toolchain/sem_ir/dump.cpp

```diff
@@ -6,305 +6,305 @@
+  out << ": " << name_scope;
+  if (name_scope.inst_id().has_value()) {
+    out << " " << file.insts().Get(name_scope.inst_id());
+  }
+  out << DumpNameIfValid(file, name_scope.name_id());
+  for (const auto& entry : name_scope.entries()) {
+    out << "\n  - " << entry.name_id << DumpNameIfValid(file, entry.name_id)
+        << ": ";
+    if (entry.result.is_poisoned()) {
+      out << "<poisoned>";
+    } else if (entry.result.is_found()) {
+      switch (entry.result.access_kind()) {
+        case AccessKind::Public:
+          out << "public ";
+          break;
+        case AccessKind::Protected:
+          out << "protected ";
+          break;
+        case AccessKind::Private:
+          out << "private ";
+          break;
       }
+      out << DumpInstSummary(file, entry.result.target_inst_id());
+    } else {
+      out << "<not-found>";
     }
-  } else {
-    llvm::errs() << '\n';
   }
+  return out.TakeStr();
 }
 
 LLVM_DUMP_METHOD auto Dump(const File& file,
-                           CompleteFacetTypeId complete_facet_type_id) -> void {
-  llvm::errs() << complete_facet_type_id << "\n";
-  if (complete_facet_type_id.has_value()) {
-    const auto& complete_facet_type =
-        file.complete_facet_types().Get(complete_facet_type_id);
-    for (auto [i, req_interface] :
-         llvm::enumerate(complete_facet_type.required_interfaces)) {
-      llvm::errs() << "  - ";
-      DumpNoNewline(file, req_interface.interface_id);
-      if (req_interface.specific_id.has_value()) {
-        llvm::errs() << "; ";
-        DumpNoNewline(file, req_interface.specific_id);
-      }
-      if (static_cast<int>(i) < complete_facet_type.num_to_impl) {
-        llvm::errs() << " (to impl)";
-      }
-      llvm::errs() << '\n';
+                           CompleteFacetTypeId complete_facet_type_id)
+    -> std::string {
+  RawStringOstream out;
+  out << complete_facet_type_id;
+  if (!complete_facet_type_id.has_value()) {
+    return out.TakeStr();
+  }
+
+  const auto& complete_facet_type =
+      file.complete_facet_types().Get(complete_facet_type_id);
+  for (auto [i, req_interface] :
+       llvm::enumerate(complete_facet_type.required_interfaces)) {
+    out << "\n  - " << Dump(file, req_interface.interface_id);
+    if (req_interface.specific_id.has_value()) {
+      out << "; " << Dump(file, req_interface.specific_id);
+    }
+    if (static_cast<int>(i) < complete_facet_type.num_to_impl) {
+      out << " (to impl)";
     }
   }
+  return out.TakeStr();
 }
 
-LLVM_DUMP_METHOD auto Dump(const File& file, SpecificId specific_id) -> void {
-  DumpNoNewline(file, specific_id);
-  llvm::errs() << '\n';
+LLVM_DUMP_METHOD auto Dump(const File& file, SpecificId specific_id)
+    -> std::string {
+  RawStringOstream out;
+  out << specific_id;
   if (specific_id.has_value()) {
-    Dump(file, file.specifics().Get(specific_id).args_id);
+    out << ": " << file.specifics().Get(specific_id);
```

Comment:
This lost dumping the args_id inst block? (which should come after a newline, not a colon)